plugins/catalog/src/components/CatalogExportButton/CatalogExportButton.tsx

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { useCallback, useEffect, useState } from 'react';
+import Button from '@material-ui/core/Button';
+import type { ButtonProps } from '@material-ui/core/Button';
+import Dialog from '@material-ui/core/Dialog';
+import DialogActions from '@material-ui/core/DialogActions';
+import DialogContent from '@material-ui/core/DialogContent';
+import DialogTitle from '@material-ui/core/DialogTitle';
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import Select from '@material-ui/core/Select';
+import Box from '@material-ui/core/Box';
+import GetAppIcon from '@material-ui/icons/GetApp';
+import {
+  alertApiRef,
+  discoveryApiRef,
+  useApi,
+} from '@backstage/core-plugin-api';
+import {
+  useBackstageStreamedDownload,
+  type BackstageStreamedDownloadOptions,
+} from './file-download';
+import { setEnabledBackendFilters } from './setEnabledBackendFilters.ts';
+import { useEntityList } from '@backstage/plugin-catalog-react';
+
+export enum CatalogExportType {
```

Comment:
Related to my future idea of custom export type registration, the supported types should be fetched from the backend